src/main/java/org/kohsuke/github/GHRepositoryForkBuilder.java

```diff
@@ -0,0 +1,144 @@
+    private final Requester req;
+    private String organization;
+    private String name;
+    private Boolean defaultBranchOnly;
+
+    static int FORK_RETRY_INTERVAL = 3000;
+
+    /**
+     * Instantiates a new Gh repository fork builder.
+     *
+     * @param repo
+     *            the repository
+     */
+    GHRepositoryForkBuilder(GHRepository repo) {
+        this.repo = repo;
+        this.req = repo.root().createRequest();
+    }
+
+    /**
+     * Sets whether to fork only the default branch.
+     *
+     * @param defaultBranchOnly
+     *            the default branch only
+     * @return the gh repository fork builder
+     */
+    public GHRepositoryForkBuilder defaultBranchOnly(boolean defaultBranchOnly) {
+        this.defaultBranchOnly = defaultBranchOnly;
+        return this;
+    }
+
+    /**
+     * Specifies the target organization for the fork.
+     *
+     * @param organization
+     *            the organization
+     * @return the gh repository fork builder
+     */
+    public GHRepositoryForkBuilder organization(GHOrganization organization) {
+        this.organization = organization.getLogin();
+        return this;
+    }
+
+    /**
+     * Sets a custom name for the forked repository.
+     *
+     * @param name
+     *            the desired repository name
+     * @return the builder
+     */
+    public GHRepositoryForkBuilder name(String name) {
+        this.name = name;
+        return this;
+    }
+
+    /**
+     * Creates the fork with the specified parameters.
+     *
+     * @return the gh repository
+     * @throws IOException
+     *             the io exception
+     */
+    public GHRepository create() throws IOException {
+        if (defaultBranchOnly != null) {
+            req.with("default_branch_only", defaultBranchOnly);
+        }
+        if (organization != null) {
+            req.with("organization", organization);
+        }
+        if (name != null) {
+            req.with("name", name);
+        }
+
+        req.method("POST").withUrlPath(repo.getApiTailUrl("forks")).send();
+
+        // this API is asynchronous. we need to wait for a bit
+        for (int i = 0; i < 10; i++) {
+            GHRepository r = lookupForkedRepository();
+            if (r != null) {
+                return r;
+            }
+            sleep(FORK_RETRY_INTERVAL);
+        }
+        throw new IOException(createTimeoutMessage());
+    }
+
+    private GHRepository lookupForkedRepository() throws IOException {
+        String repoName = name != null ? name : repo.getName();
+
+        if (organization != null) {
+            return repo.root().getOrganization(organization).getRepository(repoName);
+        }
+        return repo.root().getMyself().getRepository(repoName);
+    }
+
+    /**
+     * Sleep.
+     *
+     * @param millis
+     *            the millis
+     * @throws IOException
+     *             the io exception
+     */
+    protected void sleep(int millis) throws IOException {
```

Comment:
This should be package scoped, not protected.  

`protected` makes the method available for override externally.  You don't want that. You only want the tests to be able to override it. 
```suggestion
    void sleep(int millis) throws IOException {
```